### PR TITLE
[resolves #44] Introduce store method `shouldDehydrate` to disable dehydration for a store

### LINKED ISSUE #44: Introduce store method `shouldDehydrate` to disable dehydration for a store
This method would allow stores to control whether they should dehydrate themselves on the server to be sent to the client. We have run into times where a store may have listened for an action, but we did not need the store to be rehydrated on the client. For instance, one-time-use server-only data.

In dispatchr's `dehydrate` method:
 * check if store implements `shouldDehydrate` method
   * if method exists, call it
     * if returns `true`, call dehydrate and return store's state
     * if returns `false`, ignore the store's state completely
   * if method does not exist:
     * always dehydrate (default to true)

### PR DESCRIPTION
**Pull request by mridgway**
See #44

### CODE REVIEW

**Comment by yahoocla** [comment]
CLA is valid!

File: tests/unit/lib/Dispatcher.js [hunk 1]
```diff
@@ -209,4 +211,16 @@ describe('Dispatchr', function () {
         });
     });
 
+    describe('#shouldDehydrate', function () {
+        it('should not dehydrate stores that return false from shouldDehydrate', function () {
+            var dispatcher = new Dispatcher();
+            dispatcher.dispatch('NAVIGATE', {});
+            var state = dispatcher.dehydrate();
+            expect(state.stores).to.be.an('object');
+            console.log(state);
```
Comment (redonkulus) [inline tests/unit/lib/Dispatcher.js]: remove

**Comment by lingyan** [comment]
:ship:

**Comment by redonkulus** [comment]
:rocket:

**Comment by roderickhsiao** [comment]
:100:

**Pull request merged**